rchttp/client_test.go

```diff
@@ -148,15 +148,46 @@ func TestClientNoRetries(t *testing.T) {
 	})
 }
 
-func TestClientWithServiceTokenHasAuthHeader(t *testing.T) {
+func TestClientWithServiceAndUserPopulatesAllHeaders(t *testing.T) {
 	ts := rchttptest.NewTestServer()
 	defer ts.Close()
 	expectedCallCount := 0
 
-	Convey("Given an rchttp client with an auth token", t, func() {
+	Convey("Given an rchttp client with an auth token and context with userId", t, func() {
 		expectedAuthToken := "IAmWhoIAm"
-		// throw in a check for wrapped client instantiation
+		testUser := "hello@test"
+		upstreamRequestId := "call123"
 		httpClient := ClientWithServiceToken(ClientWithTimeout(nil, 5*time.Second), expectedAuthToken)
+		ctx := common.WithRequestId(common.SetUser(context.Background(), testUser), upstreamRequestId)
+
+		Convey("When Get() is called on a URL", func() {
+			expectedCallCount++
+			resp, err := httpClient.Get(ctx, ts.URL)
+			So(resp, ShouldNotBeNil)
+			So(err, ShouldBeNil)
+
+			call, err := unmarshallResp(resp)
+			So(err, ShouldBeNil)
+
+			Convey("Then the server sees the auth and user headers", func() {
+				So(call.CallCount, ShouldEqual, expectedCallCount)
+				So(call.Method, ShouldEqual, "GET")
+				So(call.Error, ShouldEqual, "")
+				So(call.Headers[common.AuthHeaderKey], ShouldResemble, []string{common.BearerPrefix + expectedAuthToken})
+				So(call.Headers[common.UserHeaderKey], ShouldResemble, []string{testUser})
```

Comment:
Should there be an assertion to check requestID (the upstream request ID) ?

Reply:
done